identity-verification/module_5/Metrics_for_Evaluating_an_Identity_Verification_Solution.ipynb

```diff
@@ -0,0 +1,340 @@
+    "df_similarity = df.copy()\n",
+    "df_similarity[\"SIMILARITY\"] = None\n",
+    "    \n",
+    "for index, row in df.iterrows():\n",
+    "    sourceFile = dataset_folder + row[\"SOURCE\"]\n",
+    "    targetFile = dataset_folder + row[\"TARGET\"]\n",
+    "    response_score = compare_faces(sourceFile, targetFile)\n",
+    "    df_similarity._set_value(index,\"SIMILARITY\", response_score)\n",
+    "\n",
+    "df_similarity.head()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "310c8dc5",
+   "metadata": {},
+   "source": [
+    "### BoxPlot of distribution of similarity score by test set"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e9bac38d",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sns.boxplot(data=df_similarity,\n",
+    "            x=df_similarity[\"SIMILARITY\"],\n",
+    "            y=df_similarity[\"TEST\"]).set(xlabel='Similarity Score', \n",
+    "                                         ylabel=None, \n",
+    "                                         title = \"Similarity Score Distribution\")\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "13116436",
+   "metadata": {},
+   "source": [
+    "### Descriptive Statistics"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f789400b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "df_descriptive_stats = pd.DataFrame(columns=['test','count', 'min' , 'max', 'mean', 'median', 'std'])\n",
+    "tests = [\"Genuine\", \"Imposter\"]\n",
+    "\n",
+    "for test in tests:\n",
+    "    count = df_similarity['SIMILARITY'].loc[df_similarity['TEST'] == test].count()\n",
+    "    mean = df_similarity['SIMILARITY'].loc[df_similarity['TEST'] == test].mean()\n",
+    "    max_ = df_similarity['SIMILARITY'].loc[df_similarity['TEST'] == test].max()\n",
+    "    min_ = df_similarity['SIMILARITY'].loc[df_similarity['TEST'] == test].min()\n",
+    "    median = df_similarity['SIMILARITY'].loc[df_similarity['TEST'] == test].median()\n",
+    "    std = df_similarity['SIMILARITY'].loc[df_similarity['TEST'] == test].std()\n",
+    "\n",
+    "    new_row = {'test': test,\n",
+    "                'count': count,\n",
+    "                'min': min_,\n",
+    "                'max': max_,\n",
+    "                'mean': mean,\n",
+    "                'median':median,\n",
+    "                'std': std}\n",
+    "    df_descriptive_stats = df_descriptive_stats.append(new_row,ignore_index=True)\n",
+    "\n",
+    "df_descriptive_stats"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "459d97b1",
+   "metadata": {},
+   "source": [
+    "## Step 4 - Calculate False Match and Non-Match Rates at different similarity threshold levels"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "0b71ff1a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "SimilarityThresholds = [80,85,90,95,96,97,98,99]\n",
```

Comment:
can you keep the python code consistent, using snake_case rather than PascalCase.